pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,628 @@
+  
+	# Delete a VirtualMachineExport resource
+	{{ProgramName}} vmexport delete snap1-export
+  
+	# Download a volume from an already existing VirtualMachineExport
+	{{ProgramName}} vmexport download vm1-export --volume=volume1 --output=disk.img.gz
+  
+	# Create a VirtualMachineExport and download the requested volume from it
+	{{ProgramName}} vmexport download vm1-export --create --pvc=pvc1 --volume=volume1 --output=disk.img.gz`
+
+	return usage
+}
+
+// NewVirtualMachineExportCommand returns a cobra.Command to handle the export process
+func NewVirtualMachineExportCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "vmexport",
+		Short:   "Export a VM volume.",
+		Example: usage(),
+		Args:    templates.ExactArgs("vmexport", 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			v := command{clientConfig: clientConfig}
+			return v.run(args)
+		},
+	}
+
+	cmd.Flags().StringVar(&vm, "vm", "", "Sets VirtualMachine as vmexport kind and specifies the vm name.")
+	cmd.Flags().StringVar(&snapshot, "snapshot", "", "Sets VirtualMachineSnapshot as vmexport kind and specifies the snapshot name.")
+	cmd.Flags().StringVar(&pvc, "pvc", "", "Sets PersistentVolumeClaim as vmexport kind and specifies the PVC name.")
+	cmd.MarkFlagsMutuallyExclusive("vm", "snapshot", "pvc")
+	cmd.Flags().StringVar(&outputFile, "output", "", "Specifies the output path of the volume to be downloaded.")
+	cmd.Flags().StringVar(&volumeName, "volume", "", "Specifies the volume to be downloaded.")
+	cmd.Flags().BoolVar(&shouldCreate, "create", false, "When used with the 'download' option, specifies that a VirtualMachineExport should be created from scratch.")
+	cmd.Flags().BoolVar(&insecure, "insecure", false, "When used with the 'download' option, specifies that the http request should be insecure.")
+	cmd.Flags().BoolVar(&keepvme, "keep-vme", false, "When used with the 'download' option, specifies that the vmexport object should not be deleted after the download finishes.")
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+
+	return cmd
+}
+
+// run serves as entrypoint for the vmexport command
+func (c *command) run(args []string) error {
+	if err := parseExportArguments(args); err != nil {
+		return err
+	}
+
+	namespace, _, err := c.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return fmt.Errorf("Cannot obtain KubeVirt client: %v", err)
+	}
+
+	// Finally, run the vmexport function (create|delete|download)
+	if err := exportFunction(virtClient, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// parseExportArguments parses and validates vmexport arguments and flags. These arguments should always be:
+// 	1. The vmexport function (create|delete|download)
+// 	2. The VirtualMachineExport name
+func parseExportArguments(args []string) error {
```

Comment:
it might make sense to have parseExportArguments return a struct that contains all the flags. And then pass that struct around in your functions. This will help memory dump download since they just have to populate the struct with the right information and pass it to your functions.

Reply:
I like this option, thanks!